Deprecate checkmax/max_params docs (#2739)
nf-core/website: sites/docs/src/content/docs/usage/Getting_started/configuration.md

In addition to the executor, you may find that pipeline runs occasionally fail due to a particular step of the pipeline requesting more resources than you have on your system.

To avoid these failures, you can tell Nextflow to set a cap pipeline-step resource requests against a list called `resourceLimits` specified in Nextflow config file. These should represent the maximum possible resources of a machine or node.

For example, you can place into a file the following:

```groovy
process {
  resourceLimits = [
    cpus: 32,
    memory: 256.GB,
    time: 24.h
  ]
}
```

> And supply this in your pipeline run command with `-c <custom>.conf`. Then, during a pipeline run, if for example a job exceeds the default memory request, it will be retried increasing the memory each time until either the job completes, or until it reaches a request of `256.GB`.

Reviewer: ```suggestion
And supply this in your pipeline run command with `-c <custom>.config`. Then, during a pipeline run, if for example a job exceeds the default memory request, it will be retried increasing the memory each time until either the job completes, or until it reaches a request of `256.GB`.
```

> And supply this in your pipeline run command with `-c <custom>.config`. Then, during a pipeline run, if for example a job exceeds the default memory request, it will be retried increasing the memory each time until either the job completes, or until it reaches a request of `256.GB`.

Reviewer 2: ```suggestion
And supply this in your pipeline run command with `-c <custom>.config`. Then, during a pipeline run, if, for example, a job exceeds the default memory request, it will be retried, increasing the memory each time until either the job completes or until it reaches a request of `256.GB`.
```

> Therefore, these parameters only act as a _cap_, to prevent Nextflow submitting a single job requesting resources more than what is possible on your system and requests getting out of hand.

Reviewer 2: ```suggestion
Therefore, these parameters only act as a _cap_to prevent Nextflow from submitting a single job requesting resources more than what is possible on your system and requests getting out of hand.
```

> Note that specifying these will not _increase_ the resources available to the pipeline tasks! See [Tuning workflow resources](#tuning-workflow-resources) for this.

Reviewer 2: ```suggestion
Note that specifying these will not _increase_ the resources available to the pipeline tasks! See [Tuning workflow resources](#tuning-workflow-resources) for more infomation.
```

:::note{collapse title="Note on older nf-core pipelines"}

> If you wish to use pipelines generated with the pre-`v3.0.0` nf-core template, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.

Author: ```suggestion
If you wish to use pipelines generated with the nf-core template before `v3.0.0`, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.
```

To tune workflow resources to better match your requirements, we can tweak these through [custom configuration files](#custom-configuration-files) or [shared nf-core/configs](#shared-nf-coreconfigs).

By default, most process resources are specified using process _labels_, as in the following example base config:

```groovy
process {
  resourceLimits = [
    cpus: 32,
    memory: 256.GB,
    time: 24.h
  ]
  withLabel:process_low {
    cpus   = { 2 * task.attempt }
    memory = { 14.GB * task.attempt }
    time   = { 6.h  * task.attempt }
  }
  withLabel:process_medium {
    cpus   = { 6  * task.attempt }
    memory = { 42.GB * task.attempt }
    time   = { 8.h * task.attempt }
  }
  withLabel:process_high {
    cpus   = { 12 * task.attempt }
    memory = { 84.GB * task.attempt }
    time   = { 10.h * task.attempt }
  }
}
```

> The `resourceLimits` list sets absolute maximums any pipeline job can request (typically corresponding to the maximum available resources on your machine). The label blocks indicate the initial 'default' resources a pipeline job will request. For most nf-core pipelines, if a job runs out of memory, it will re-try the job but increasing the amount resource requested up to the `resourceLimits` maximum.

Reviewer 2: ```suggestion
The `resourceLimits` list sets the absolute maximums any pipeline job can request (typically corresponding to the maximum available resources on your machine). The label blocks indicate the initial 'default' resources a pipeline job will request. For most nf-core pipelines, if a job runs out of memory, it will retry the job but increase the amount of resource requested up to the `resourceLimits` maximum.
```

:::note{collapse title="Note on older nf-core pipelines"}

> If you wish to use pipelines generated with the pre-`v3.0.0` nf-core template, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.

Author: ```suggestion
If you wish to use pipelines generated with the nf-core template before `v3.0.0`, and/or when running with Nextflow versions earlier than 24.04.0 you may need use a different syntax to prevent the resources from exceeding a maximum limit.
```

}
```

> You can also be more specific than this by targeting a given _process_ (job) name instead of it's label using `withName`. You can see the process names in your console log when the pipeline is running For example:

Reviewer 2: ```suggestion
You can also be more specific than this by targeting a given _process_ (job) name instead of its label using `withName`. You can see the process names in your console log when the pipeline is running For example:
```
